packages/grpc-js/src/server.ts

```diff
@@ -1421,41 +1420,72 @@ export class Server {
       }
 
       if (this.keepaliveTimeMs < KEEPALIVE_MAX_TIME_MS) {
-        keeapliveTimeTimer = setInterval(() => {
-          keepaliveTimeoutTimer = setTimeout(() => {
-            sessionClosedByServer = true;
-            session.close();
+        keepaliveInterval = setInterval(() => {
+          const keepaliveTimeout = setTimeout(() => {
+            if (keepaliveInterval) {
+              clearInterval(keepaliveInterval);
+              keepaliveInterval = null;
```

Comment:
These cleanup actions should be handled in the `close` event handler triggered by the `session.close` call below. We don't also need to do them here.